File: app/src/main/java/com/iug/coursehub/data/local/db/dao/LessonDao.java
package com.iug.coursehub.data.local.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Query;
import androidx.room.Upsert;

import com.iug.coursehub.data.local.db.entity.Lesson;

import java.util.List;

@Dao
public interface LessonDao {

    @Upsert
    void upsertLesson(Lesson lesson);

    @Query("SELECT * FROM lesson WHERE id = :id")
    Lesson getLessonById(int id);

    @Query("SELECT * FROM lesson WHERE courseId = :courseId")
    LiveData<List<Lesson>> getAllLessonsOfCourse(int courseId);

    @Delete
    void deleteLesson(Lesson lesson);

    @Query("SELECT SUM(id) FROM lesson WHERE courseId = :courseId")
    int getNoOfLessonsByCourseId(int courseId);
}

File: app/src/main/java/com/iug/coursehub/data/local/db/dao/LessonDao.kt
package com.iug.coursehub.data.local.db.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Query
import androidx.room.Upsert
import com.iug.coursehub.data.local.db.entity.Course
import com.iug.coursehub.data.local.db.entity.Lesson

@Dao
interface LessonDao {

    @Upsert
    suspend fun upsertLesson(lesson: Lesson)

    @Query("SELECT * FROM lesson WHERE id = :id")
    suspend fun getLessonById(id: Int): Lesson

    @Query("SELECT * FROM lesson WHERE courseId = :courseId")
    fun getAllLessonsOfCourse(courseId: Int): LiveData<List<Lesson>>

    @Delete
    suspend fun deleteLesson(lesson: Lesson)

    @Query("SELECT SUM(id) FROM lesson WHERE courseId = :courseId")
    fun getNoOfLessonsByCourseId(courseId: Int): Int
}